blocks/library/more/index.js

```diff
@@ -68,33 +68,54 @@ export const settings = {
 		],
 	},
 
-	edit( { attributes, setAttributes, isSelected } ) {
-		const { customText, noTeaser } = attributes;
-
-		const toggleNoTeaser = () => setAttributes( { noTeaser: ! noTeaser } );
-		const defaultText = __( 'Read more' );
-		const value = customText !== undefined ? customText : defaultText;
-		const inputLength = value.length ? value.length + 1 : 1;
-
-		return [
-			isSelected && (
-				<InspectorControls key="inspector">
-					<ToggleControl
-						label={ __( 'Hide the teaser before the "More" tag' ) }
-						checked={ !! noTeaser }
-						onChange={ toggleNoTeaser }
+	edit: class extends Component {
+		constructor() {
+			super( ...arguments );
+			this.onChangeInput = this.onChangeInput.bind( this );
+
+			this.state = {
+				defaultText: __( 'Read more' ),
+			};
+		}
+
+		onChangeInput( event ) {
+			this.setState( {
+				defaultText: '',
+			} );
+
+			const value = event.target.value.length === 0 ? undefined : event.target.value;
+			this.props.setAttributes( { customText: value } );
+		}
+
+		render() {
+			const { customText, noTeaser } = this.props.attributes;
+			const { setAttributes, isSelected } = this.props;
+
+			const toggleNoTeaser = () => setAttributes( { noTeaser: ! noTeaser } );
+			const defaultText = this.state.defaultText;
```

Comment:
Minor: Personal preference: ES2015 object destructuring is perfect for these cases where assigning a variable the same name as the property of an object:

```js
const { defaultText } = this.state;
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring

It's the same pattern as above for `const { setAttributes, isSelected } = this.props;`, which could have been expressed as:

```js
const setAttributes = this.props.setAttributes;
const isSelected = this.props.isSelected;
```

Note that it's particularly useful to avoid verbosity when destructuring multiple properties (which, who knows, might happen here some day).